tools/bazar/actions/BazarListeAction.php

```diff
@@ -179,84 +195,50 @@ public function run()
                 && (!isset($this->arguments['calledBy']) || $this->arguments['calledBy'] !== 'CalendrierAction')) {
             return $this->callAction('calendrier', $this->arguments);
         }
-
-        $entryManager = $this->getService(EntryManager::class);
-        $formManager = $this->getService(FormManager::class);
-        $externalWikiService = $this->getService(ExternalBazarService::class);
-
-        if (!isset($GLOBALS['_BAZAR_']['nbbazarliste'])) {
-            $GLOBALS['_BAZAR_']['nbbazarliste'] = 0;
-        }
-        ++$GLOBALS['_BAZAR_']['nbbazarliste'];
-
-        // TODO put in all bazar templates
-        $this->wiki->AddJavascriptFile('tools/bazar/libs/bazar.js');
-
-        // External mode activated ?
-        if ($this->arguments['externalModeActivated']) {
-            $forms = $externalWikiService->getFormsForBazarListe($this->arguments['externalIds'], $this->arguments['refresh']);
-            $entries = $externalWikiService->getEntries([
-                'forms' => $forms,
-                'refresh' => $this->arguments['refresh'],
-                'queries' => $this->arguments['query']
+        
+        $bazarListService = $this->getService(BazarListService::class);
+        $bazarListService->setArguments($this->arguments);
```

Comment:
Like BazarListeService is a service, I prefer that its state does not depends of arguments.
So it would be preferable that `$this->arguments` was a parameter of ` $bazarListService->getForms()` and nothing was stored in BazarListeService because it is a service.

Reply:
Je pense pas qu'il y ait de soucis pour un service de stocker des choses, si c'est pour sa tambouille interne? Mais là effectivement c'est un peu bizarre de faire le `->setArguments` avant d'appeler les autres méthodes, donc j'ai fait la modif comme suggéré ! merci :)